src/main/java/org/apache/sling/resourceresolver/impl/mapping/ResourceMapperImpl.java

```diff
@@ -72,7 +72,7 @@ public String getMapping(String resourcePath, HttpServletRequest request) {
         
         Collection<String> mappings = getAllMappings(resourcePath, request);
         if ( mappings.isEmpty() )
-            return null;
+            return "";
```

Comment:
I would prefer that we handle this differently, in the `getAllMappings` method. The javadoc of that mapping promises that no empty results will be returned.

https://github.com/apache/sling-org-apache-sling-api/blob/org.apache.sling.api-2.24.0/src/main/java/org/apache/sling/api/resource/mapping/ResourceMapper.java#L109

Reply:
Finding an appropriate mappings relies on having non-empty mappedPath or non-null nonDecoratedResource but in case if mappedPath is empty and nonDecoratedResource is null (it could be possible based on doc [0]) then there could be no possible mapping and the result would be empty. Besides I cold not think of any possible mapping which ought to be returned in such case, any thoughts ?

[0]: https://github.com/apache/sling-org-apache-sling-resourceresolver/blob/deb25ba47cf563e5bc52978eef36e80d735a067c/src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverImpl.java#L718